### PR TITLE
cyrusdb_skiplist: use xsyslog for errors

### PR DESCRIPTION
**Pull request by elliefm**
More of this.  As usual, please try building this too, in case your compiler finds something mine didn't.  Thanks!

### CODE REVIEW

File: lib/cyrusdb_skiplist.c [hunk 1]
```diff
@@ -1336,7 +1350,8 @@ static int mystore(struct dbengine *db,
     lseek(tid->syncfd, tid->logend, SEEK_SET);
     r = retry_writev(tid->syncfd, iov, num_iov);
     if (r < 0) {
-        syslog(LOG_ERR, "DBERROR: retry_writev(): %m");
+        /* XXX is there a filename or something we can log here */
+        xsyslog(LOG_ERR, "DBERROR: retry_writev failed", NULL);
```
Comment (ksmurchison) [inline lib/cyrusdb_skiplist.c:1354]: I think we can use db->fname here

File: lib/cyrusdb_skiplist.c [hunk 2]
```diff
@@ -1427,7 +1442,8 @@ static int mydelete(struct dbengine *db,
         /* update end-of-log */
         r = retry_write(tid->syncfd, (char *) writebuf, 8);
         if (r < 0) {
-            syslog(LOG_ERR, "DBERROR: retry_write(): %m");
+            /* XXX is there a filename or something we can log here */
+            xsyslog(LOG_ERR, "DBERROR: retry_write failed", NULL);
```
Comment (ksmurchison) [inline lib/cyrusdb_skiplist.c:1446]: I think we can use db->fname here

**Review by ksmurchison — CHANGES_REQUESTED** [review]
This looks good (pending my 2 inline comments) and compiles cleanly for me

**Comment by elliefm** [comment]
Thanks.  I've updated those two lines to log the db->fname.  Trivial changes, but we're also not in any hurry here... can you give it another compile check once you're back from break, just in case?

**Comment by brong** [comment]
Gonna add this to our build, assuming that's all happy we'll merge it :)

**Review by rsto — COMMENTED** [review]
Looks good to me.

**Pull request merged**